google/cloud/storage/client.py

```diff
@@ -565,11 +605,27 @@ def download_blob_to_file(self, blob_or_uri, file_obj, start=None, end=None):
 
 
         """
+        if not isinstance(blob_or_uri, Blob):
+            blob_or_uri = Blob.from_string(blob_or_uri)
+
+        download_url = blob_or_uri._get_download_url(
+            self,
+            if_generation_match=if_generation_match,
+            if_generation_not_match=if_generation_not_match,
+            if_metageneration_match=if_metageneration_match,
+            if_metageneration_not_match=if_metageneration_not_match,
+        )
+        headers = _get_encryption_headers(blob_or_uri._encryption_key)
+        headers["accept-encoding"] = "gzip"
+
+        transport = self._http
```

Comment:
We can omit this `transport` var and just put `self._http` to `_do_download()` call